packages/opentelemetry-core/src/trace/sampler/TraceIdRatioBasedSampler.ts

```diff
@@ -16,21 +16,22 @@
 
 import { Sampler, SamplingDecision, SamplingResult } from '@opentelemetry/api';
 
+const SAMPLE_HEX_LEN = 8;
+
 /** Sampler that samples a given fraction of traces based of trace id deterministically. */
 export class TraceIdRatioBasedSampler implements Sampler {
+  private _upperBound: number;
+
   constructor(private readonly _ratio: number = 0) {
     this._ratio = this._normalize(_ratio);
+    this._upperBound = Math.floor(this._ratio * 16 ** SAMPLE_HEX_LEN);
   }
 
   shouldSample(context: unknown, traceId: string): SamplingResult {
-    let accumulation = 0;
-    for (let idx = 0; idx < traceId.length; idx++) {
-      accumulation += traceId.charCodeAt(idx);
-    }
-    const cmp = (accumulation % 100) / 100;
+    const value = parseInt(traceId.slice(0, SAMPLE_HEX_LEN), 16);
```

Comment:
In the [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#retrieving-the-traceid-and-spanid), there is no assumption that the first 8 characters of the hex representation of the trace id should be randomly generated. I've seen usage that generates trace ids in one replica with the same prefix but with a different suffix. If the sampler only taps on the first 8 characters of trace ids, it would fail on those implementations.

Reply:
Thanks for telling me the same prefix use case.
I've updated my code to accumulate the whole TraceId to decide should sample.